lib/usdUfe/ufe/UsdClipboardHandler.cpp

```diff
@@ -0,0 +1,212 @@
+{
+    auto usdItem = std::dynamic_pointer_cast<UsdSceneItem>(item);
+    if (!usdItem)
+        return false;
+
+    const auto prim = usdItem->prim();
+    if (!prim)
+        return false;
+
+    const auto primParent = prim.GetParent();
+    if (!primParent)
+        return false;
+
+    // Special cut conditions that only applies to UsdShadeNodeGraph and UsdShadeShader.
+    if (PXR_NS::UsdShadeNodeGraph(prim) || PXR_NS::UsdShadeShader(prim)) {
+        const auto primAttrs = prim.GetAuthoredAttributes();
+        for (const auto& attr : primAttrs) {
+            const auto kBaseNameAndType
+                = PXR_NS::UsdShadeUtils::GetBaseNameAndType(PXR_NS::TfToken(attr.GetName()));
+
+            if (kBaseNameAndType.second != PXR_NS::UsdShadeAttributeType::Output
+                && kBaseNameAndType.second != PXR_NS::UsdShadeAttributeType::Input) {
+                continue;
+            }
+
+            if (kBaseNameAndType.second == PXR_NS::UsdShadeAttributeType::Input) {
+                // The attribute could be a destination for connected sources, so check for its
+                // connections.
+                PXR_NS::UsdShadeSourceInfoVector sourcesInfo
+                    = pxr::UsdShadeConnectableAPI::GetConnectedSources(attr);
+
+                if (!sourcesInfo.empty()) {
+                    return false;
+                }
+            }
+
+            if (kBaseNameAndType.second == PXR_NS::UsdShadeAttributeType::Output) {
+                // The attribute could be a source connection, we have to explore the siblings.
+                for (auto&& child : primParent.GetChildren()) {
+                    if (child == prim) {
+                        continue;
+                    }
+
+                    for (const auto& otherAttr : child.GetAttributes()) {
+                        const auto childAttrBaseNameAndType
+                            = PXR_NS::UsdShadeUtils::GetBaseNameAndType(
+                                PXR_NS::TfToken(otherAttr.GetName()));
+
+                        if (childAttrBaseNameAndType.second == PXR_NS::UsdShadeAttributeType::Input
+                            && isAttributeConnected(attr, otherAttr)) { }
+                        return false;
+                    }
+                }
+            }
+
+            // Check also if there are connections to the parent.
+            for (const auto& otherAttr : primParent.GetAttributes()) {
+                const auto parentAttrBaseNameAndType = PXR_NS::UsdShadeUtils::GetBaseNameAndType(
+                    PXR_NS::TfToken(otherAttr.GetName()));
+
+                if (parentAttrBaseNameAndType.second == PXR_NS::UsdShadeAttributeType::Output
+                    && isAttributeConnected(attr, otherAttr)) {
+                    return false;
+                }
+            }
+        }
+
+        // If there are no connections, then the item can be cut.
+        return true;
+    }
+
+    try {
+        enforceMutedLayer(prim, "remove");
```

Comment:
I thought there was a no-throw version that returned a value, but maybe I mis-remember.

Reply:
Yes, enforce just calls has, so it would work.